lib/versioning/VersionID.js

```diff
@@ -181,9 +188,15 @@ function base62Encode(str) {
     const enc1 = base62Integer.encode(part1);
     const enc2 = base62Integer.encode(part2);
     const enc3 = base62String.encode(part3);
-    return (B62V_EPAD + enc1).slice(-B62V_EPAD.length) +
-        (B62V_EPAD + enc2).slice(-B62V_EPAD.length) +
-        enc3;
+    const padded1 = (B62V_EPAD + enc1).slice(-B62V_EPAD.length);
+    const padded2 = (B62V_EPAD + enc2).slice(-B62V_EPAD.length);
+    const enc3String = enc3.toString();
+    const totalLength = padded1.toString().length +
+        padded2.toString().length +
+        enc3String.length;
+    const enc3PaddingNeeded = 32 - totalLength + enc3String.length;
+
+    return padded1 + padded2 + enc3.toString().padEnd(enc3PaddingNeeded, '\u0000');
```

Comment:
I don't think we can return null bytes in the encoded version: it should only contain base62 characters. Even if we are allowed other characters than base62 (letters and digits) we have to consider what the application would do with such characters, it's surely calling for issues to include nonprintable characters in the exposed version ID.

Reply:
I think the only way then is to rely on the fact that the unpadded string should be 22 bytes and then add noise. Then decoding is a matter of stripping the last 10 bytes

Reply:
@vrancurel thoughts on this?

Reply:
oh right I though we were padding null bytes in the decoded version. Yes the encoded key shall only return base62 characters.